packages/client/src/comp/util/Signature.tsx

```diff
@@ -18,11 +20,13 @@ type Props = React.PropsWithChildren<{
 }>
 
 export const Signature: React.FC<Props> = ({ setSignature }) => {
+  const { search } = useLocation()
+  const params = parseQS(search)
   return <Margin>
-    <Switchable>
+    <Switchable visible={Boolean(!params.case)}>
     {
       (choice: Choice) => {
-        if (choice === 'canvas') {
+        if (choice === 'canvas' || params.case === 'a') {
```

Comment:
Can we name the test cases better?  Maybe they can be 'both' or 'upload'?  We are testing whether they can be both Canvas and Upload or just Upload.  I thin you have the logic wrong here.

Reply:
done